File: software/examples/vbnet/ExampleSimple.vb
Imports System
Imports Tinkerforge

Module ExampleSimple
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your UV Light Bricklet 2.0

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim uvl As New BrickletUVLightV2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get current UV-A
        Dim uva As Integer = uvl.GetUVA()
        Console.WriteLine("UV-A: " + (uva/10.0).ToString() + " mW/m²")

        ' Get current UV-B
        Dim uvb As Integer = uvl.GetUVB()
        Console.WriteLine("UV-B: " + (uvb/10.0).ToString() + " mW/m²")

        ' Get current UV index
        Dim uvi As Integer = uvl.GetUVI()
        Console.WriteLine("UV Index: " + (uvi/10.0).ToString())

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
	"github.com/tinkerforge/go-api-bindings/uv_light_v2_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your UV Light Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	uvl, _ := uv_light_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current UV-A.
	uva, _ := uvl.GetUVA()
	fmt.Printf("UV-A: %f mW/m²\n", float64(uva)/10.0)

	// Get current UV-B.
	uvb, _ := uvl.GetUVB()
	fmt.Printf("UV-B: %f mW/m²\n", float64(uvb)/10.0)

	// Get current UV index.
	uvi, _ := uvl.GetUVI()
	fmt.Printf("UV Index: %f\n", float64(uvi)/10.0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
